interfaces.go

```diff
@@ -156,6 +156,20 @@ type FilterQuery struct {
 	Topics [][]common.Hash
 }
 
+// WorkShareCriteria represents the parameters for a work share filter.
+type WorkShareCriteria struct {
+	QuaiCoinbase    common.Address `json:"quaiCoinbase"`    // the Quai coinbase address of the work share
```

Comment:
i think we want to make sure that these values are good, i think more of a sanity checks on all these fields